src/main.c

```diff
@@ -44,16 +49,51 @@ static long get_til_next_millis(long last_timestamp) {
 }
 
 static int luasnowflake_init(lua_State *L) {
+    conf.snowflake_epoc = luaL_optlong(L, 3, SNOWFLAKE_EPOC);
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    if (conf.snowflake_epoc < 0 || conf.snowflake_epoc > tv.tv_sec * 1000) {
+        char buffer[32];
+        snprintf(buffer, 32, "%ld", tv.tv_sec * 1000);
+        return luaL_error(L, "snowflake_epoc must be an long n where 1 ≤ n ≤ %s", buffer);
+    }
+
+    conf.node_id_bits = luaL_optint(L, 4, NODE_ID_BITS);
+    if (conf.node_id_bits < 1) {
+        return luaL_error(L, "node_id_bits must be an integer n where ≥ 1");
+    }
+
+    conf.datacenter_id_bits = luaL_optint(L, 5, DATACENTER_ID_BITS);
+    if (conf.datacenter_id_bits < 1) {
+        return luaL_error(L, "datacenter_id_bits must be an integer n where ≥ 1");
+    }
+
+    conf.sequence_bits = luaL_optint(L, 6, SEQUENCE_BITS);
+    if (conf.sequence_bits < 1) {
+        return luaL_error(L, "sequence_bits must be an integer n where ≥ 1");
+    }
+
+    if ((conf.node_id_bits + conf.datacenter_id_bits + conf.sequence_bits) > 32) {
```

Comment:
Could you explain why we need the timestamp >= 32 bits

Reply:
The timestamp >= 32 is because it is in milliseconds and only lasts 49 days when it equals 32 bits

Reply:
I see. Can you add it as a comment?

Reply:
added